doc/developer-guides/hld/ivshmem-hld.rst

```diff
@@ -181,3 +181,57 @@ post-launched VMs (VM1 and VM2).
+************************************************
+
+As previously highlighted, ACRN 2.0 provides the capability to create shared memory regions between Post-Launch user VMs known as “Inter-VM Communication”. This mechanism is based on ivshmem v1.0 exposing virtual PCI devices for the shared regions (in SOS memory for this release). This feature is just an adoption of a community approved design for shared memory between VMs, following same specification for KVM/QEMU (`Link <https://git.qemu.org/?p=qemu.git;a=blob_plain;f=docs/specs/ivshmem-spec.txt;hb=HEAD>`_).
+Following ACRN Threat Model, the policy definition regarding the allocation and assignment of these regions is controlled by the SOS, which is part of ACRN’s Trusted Computing Base (TCB). However, to secure inter-VM communication between any userspace applications that harness this channel, applications will face more requirements for the Confidentiality, Integrity, and Authenticity of shared/transferred data. It is the application’s development team responsibility to define a threat model/security architecture for the application and utilize custom/public libraries and toolkits accordingly.
+In this document we provide an overview about potential hardening techniques from user space applications perspective. Please consider those pointers when defining the security architecture and threat model for your application.
+
+.. note:: This not a definitive guide on the security technologies to be used or how to implement security. We provide general pointers not bounded to a specific OS or use-case.
+
+1. **Secure Feature Configurability**
+
+   - ACRN ensure a minimal control plane for the configuration of the memory regions boundaries and name handles. This is managed only by the SOS during the creation of the guest VM through the Device Model (DM).
+   - SOS Admin should refer to the usage guide for secure usage of the configuration flow.
+   - Create different permission/groups for ```admin``` role to segregate between other entity that might have access to SOS. For example, only admin permissions allow R/W/X on DM binary.
+   - Reference:  `Inter-VM communication Usage guide <https://projectacrn.github.io/latest/developer-guides/hld/ivshmem-hld.html>`_
+
+
+2. **Apply Access Control**
+
+   - Add restrictions based on behavior or subject/object rules around information flow and accesses.
+   - In SOS, consider the ```/dev/shm``` device node as a critical interface with special access requirement. Those requirements can be fulfilled using any of the existing opensource MAC technologies or even ACLs depending on the OS compatibility (Clear, Ubuntu, windows, etc..) and integration/management complexity.
+   - In UOS, the shared memory region can be accessed using ```mmap()``` of UIO device node. Please consider applying access control accordingly. Other complementary info can be found under:
+      * ```/sys/class/uio/uioX/device/resource2``` --> shared memory base address.
```

Comment:
Add a blank line before this first item of a list

Reply:
done